pkg/sync/syncer.go

```diff
@@ -1032,3 +1040,34 @@ func NewSyncer(ctx context.Context, c types.ConnectorClient, opts ...SyncOpt) (S
 
 	return s, nil
 }
+
+// This method checks if the principal resource already exists. If it does not, we should put the resource.
+// The return value indicates if the resource was put or not by the boolean value.
+func (s *syncer) ensureResourceExistence(ctx context.Context, principal *v2.Resource) error {
+	l := ctxzap.Extract(ctx)
+	// Check if the principal of the grant already exists as a resource. If it does not, we should jit the resource.
+	_, err := s.store.GetResource(ctx, &reader_v2.ResourcesReaderServiceGetResourceRequest{
+		ResourceId: &v2.ResourceId{
+			ResourceType: principal.Id.ResourceType,
+			Resource:     principal.Id.Resource,
+		},
+	})
+	if err != nil {
```

Comment:
I think I'd write this as `err != nil && !errors.Is(err, sql.ErrNoRows)` to save an indent level